src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -239,6 +239,19 @@ private function parseDefinition(\DOMElement $service, $file)
             $definition->addAutowiringType($type->textContent);
         }
 
+        $autowireTags = array();
+        foreach ($this->getChildren($service, 'autowire') as $type) {
+            $autowireTags[] = $type->textContent;
+        }
+
+        if (!empty($autowireTags)) {
```

Comment:
Even better: `if ($autowiredTags) {`
I'm fighting against the idea that using count() provides any type information. It does not.